app/helpers/strong_parameters_helper.rb

```diff
@@ -0,0 +1,98 @@
+module StrongParametersHelper
+  def permitted_attributes
+    Foreman::PermittedAttributes
+  end
+
+  def foreman_params
+    permitted_params = "permitted_#{controller_name.singularize}_attributes"
+    raise Foreman::Exception.new(N_('Can not find parameter method')) unless self.respond_to?(permitted_params.to_sym)
+    params.require(controller_name.singularize.to_sym).permit(*send(permitted_params))
+  end
+
+  delegate(*(Foreman::PermittedAttributes::ATTRIBUTES + [{:to => :permitted_attributes, :prefix => :permitted}]))
+
+
+  def permitted_operatingsystem_attributes
+    permitted_attributes.operatingsystem_attributes +
+        [
+          :os_parameters_attributes => permitted_os_parameters_attributes,
+          :os_default_templates_attributes => permitted_os_default_templates_attributes
+        ]
+  end
+
+  def permitted_user_attributes
+    if editing_self?
```

Comment:
Please use a guard clause instead of wrapping the whole method in an if else clause.